velox/dwio/common/Retry.h

```diff
@@ -86,26 +88,32 @@ class ExponentialBackoff : public IRetryPolicy {
   ExponentialBackoff(
       RetryDuration start,
       RetryDuration max,
-      uint64_t maxRetries = std::numeric_limits<uint64_t>::max(),
-      RetryDuration maxTotal = RetryDuration::zero())
+      uint64_t maxRetries,
+      RetryDuration maxTotal,
+      bool countExecutionTime)
       : maxWait_(max),
         maxTotal_(maxTotal),
+        countExecutionTime_(countExecutionTime),
         nextWait_(start),
-        total_(0),
+        totalWait_(0),
         retriesLeft_(maxRetries) {
     DWIO_ENSURE_LE(start.count(), max.count());
     DWIO_ENSURE(maxTotal_.count() == 0 || maxTotal_.count() > start.count());
   }
 
-  folly::Optional<RetryDuration> nextWaitTime() {
-    if (retriesLeft_ == 0 || (maxTotal_.count() > 0 && total_ >= maxTotal_)) {
+  void start() override {
+    startTime_ = std::chrono::system_clock::now();
```

Comment:
Can we set this in ctor or do we expect a long delay between retry policy ctor and the start of the first io?

Reply:
From the interface there is no guarantee the delay is small so better not have this assumption.  Otherwise we should have `withRetry` taking factory instead of the policy instance.